internal/lang/funcs/filesystem.go

```diff
@@ -197,16 +199,26 @@ func makeTemplateFileFuncImpl(baseDir string, funcsCb func() map[string]function
 		},
 		Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
 			pathArg, pathMarks := args[0].Unmark()
-			expr, err := loadTmpl(pathArg.AsString(), pathMarks)
+			expr, marks, err := loadTmpl(pathArg.AsString(), pathMarks)
 			if err != nil {
 				return cty.DynamicVal, err
 			}
 
 			result, err := renderTemplate(expr, args[1], funcsCbDepth())
-			return result.WithMarks(pathMarks), err
+
+			// Now we combine pathMarks and template Marks to ensure that our result is
```

Comment:
Reviewers: Please check if we need to combine the marks here for the path/etc or if we can just return `result.WithMarks(marks)` instead